cordformation/src/main/kotlin/net/corda/plugins/DockerImage.kt

```diff
@@ -0,0 +1,133 @@
+package net.corda.plugins
+
+import org.gradle.api.DefaultTask
+import com.spotify.docker.client.DefaultDockerClient
+import org.gradle.api.file.*
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.tasks.*
+import java.io.File
+import java.nio.file.Path
+import java.nio.file.Paths
+import javax.inject.Inject
+
+
+open class DockerImage  @Inject constructor(objects: ObjectFactory, layouts: ProjectLayout) : DefaultTask() {
+
+    private companion object{
+        //Currently a bug in gradle's projectFile that writes the file with a lower case 'd'
+        //using this name ensures that the docker plugin can find the file as expected.
+        private const val DOCKER_FILE_NAME = "dockerfile"
+    }
+
+    init {
+        description = "Creates a docker file and immediately builds that image to the local repository."
+        group = "CordaDockerDeployment"
+    }
+
+    @get:Input
+    internal var baseImage: String? = null
+
+    fun baseImage(baseImage: String) {
+        this.baseImage = baseImage
+    }
+
+    @get:Optional
+    @get:InputFile
+    internal var trustRootStoreFile: File? = null
+
+    fun trustRootStoreFile(trustRootStoreFile: File) {
+        this.trustRootStoreFile = trustRootStoreFile
+    }
+
+    @get:Optional
+    @get:Input
+    internal var buildImage: Boolean = true
+
+    fun buildImage(doBuild: Boolean) {
+        this.buildImage = doBuild
+    }
+
+    @get:Optional
+    @get:Input
+    internal var dockerImageTag: String? = null
+
+    fun dockerImageTag(tag: String) {
+        dockerImageTag = tag
+    }
+
+    private val _jars: ConfigurableFileCollection = project.files()
+
+    @get:InputFiles
+    @get:SkipWhenEmpty
+    val cordaJars: FileCollection get() = _jars
```

Comment:
It's used. Code breaks if you remove it. Simplest Test :P

Reply:
I mean that I couldn't see anything iterating over the contents of `_jars` / `cordaJars`, and yet this looked like the property that you'd most want to iterate over.

Reply:
Wait, are you saying it's not used in the task body or not used in the plugin at all ? 

I don't need to iterate over them really, i just need to shovel those bad boys into the build docker folder as they are so that docker can find them when we build it in the task.

Reply:
I was talking about in the `@TaskAction`. However, your recent commits have changed that :heavy_check_mark:.